docs/public/specifications/test-zones/Delegation-TP/delegation03.md

```diff
@@ -0,0 +1,122 @@
+:-----------------------------|:-----------------------------------------|:-------------------------------------------
+REFERRAL-SIZE-OK-1            | REFERRAL_SIZE_OK                         | 2)
+REFERRAL-SIZE-OK-2            | REFERRAL_SIZE_OK                         | 2)
+REFERRAL-SIZE-TOO-LARGE-1     | REFERRAL_SIZE_TOO_LARGE                  | 2)
+REFERRAL-SIZE-TOO-LARGE-2     | REFERRAL_SIZE_TOO_LARGE                  | 2)
+
+1\) All tags except for those specified as "Forbidden message tags" (no
+  instances for these test scenarios)
+
+2\) All tags except for those specified as "Mandatory message tags"
+
+
+## Test zone setup
+
+Assumptions for the scenario specifications unless otherwise specified for the
+specific scenario:
+* For each scenario zone there are two name server configured.
+  * Both name servers are in-bailiwick
+  * Both name servers have both IPv4 and IPv6 addresses
+  * All addresses are distinct
+  * All required glue are present in the delegation.
+  * There is no actual zone or zone file, only a delegation.
+    * For this test, that the zone is missing is disregarded.
+
+### REFERRAL-SIZE-OK-1
+This is the happy path.
+
+* Zone: referral-size-ok-1.delegation03.xa.
+
+### REFERRAL-SIZE-OK-2
+Referral is large, but not too large. The name servers are in-bailiwick.
+
+* Zone: referral-size-ok-2.delegation03.xa.
+  * ns1 is "ns1.ipv4-large-but-not-too-large.referral-size-ok-2.delegation03.xa".
+  * ns1 is "ns1.ipv6-large-but-not-too-large.referral-size-ok-2.delegation03.xa".
+  * ns2 is "ns2.ipv4-large-but-not-too-large.referral-size-ok-2.delegation03.xa".
+  * ns2 is "ns2.ipv6-large-but-not-too-large.referral-size-ok-2.delegation03.xa".
+
+### REFERRAL-SIZE-TOO-LARGE-1
+Referal is too large and name servers are in-bailiwick
+
+* Zone: referral-size-too-large-1.delegation03.xa
+  * name server names are relative the zone name:
+    * ns1 is "ns1.1abcdefghijklmnopqrstuv.1defghijkl"
+    * ns2 is "ns2.2abcdefghijklmnopqrstuv.2defghijkl"
+    * ns3 is "ns3.2abcdefghijklmnopqrstuv.3defghijkl"
+    * ns4 is "ns4.2abcdefghijklmnopqrstuv.4defghijkl"
+    * ns5 is "ns5.2abcdefghijklmnopqrstuv.5defghijkl"
+
+
+### REFERRAL-SIZE-TOO-LARGE-2
+Referal is too large and name servers are out-of-bailiwick with no glue.
+
+* Zone: referral-size-too-large-2.delegation03.xa
+  * The zone is delegated to ns1, ns2, ns3 and ns4.
+    * ns1 is "ns1.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxy.referral-size-too-large-2.delegation03.xb"
+    * ns2 is "ns2.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxy.referral-size-too-large-2.delegation03.xb"
+    * ns3 is "ns3.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxy.referral-size-too-large-2.delegation03.xb"
+    * ns4 is "ns4.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxy.referral-size-too-large-2.delegation03.xb"
```

Comment:
Authoritative name servers might be able to use name compression for the second, third and fourth name server name. It’s probably better to follow a similar naming scheme as for the name server names in REFERRAL-SIZE-TOO-LARGE-1.

Reply:
Fixed.